receiver/awscontainerinsightreceiver/internal/neuron/neuron_monitor_scraper_config.go

```diff
@@ -0,0 +1,106 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package neuron // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/neuron"
+
+import (
+	"os"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/discovery/kubernetes"
+	"github.com/prometheus/prometheus/model/relabel"
+
+	ci "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/containerinsight"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/prometheusscraper"
+)
+
+const (
+	collectionInterval        = 60 * time.Second
+	jobName                   = "containerInsightsNeuronMonitorScraper"
+	scraperMetricsPath        = "/metrics"
+	scraperK8sServiceSelector = "k8s-app=neuron-monitor-service"
+)
+
+func GetNeuronScrapeConfig(hostinfo prometheusscraper.HostInfoProvider) *config.ScrapeConfig {
+
+	return &config.ScrapeConfig{
+		ScrapeInterval: model.Duration(collectionInterval),
+		ScrapeTimeout:  model.Duration(collectionInterval),
+		JobName:        jobName,
+		Scheme:         "http",
+		MetricsPath:    scraperMetricsPath,
+		ServiceDiscoveryConfigs: discovery.Configs{
+			&kubernetes.SDConfig{
+				Role: kubernetes.RoleService,
+				NamespaceDiscovery: kubernetes.NamespaceDiscovery{
+					IncludeOwnNamespace: true,
+				},
+				Selectors: []kubernetes.SelectorConfig{
+					{
+						Role:  kubernetes.RoleService,
+						Label: scraperK8sServiceSelector,
+					},
+				},
+			},
+		},
+		MetricRelabelConfigs: GetNeuronMetricRelabelConfigs(hostinfo),
+	}
+}
+
+func GetNeuronMetricRelabelConfigs(hostinfo prometheusscraper.HostInfoProvider) []*relabel.Config {
+
+	return []*relabel.Config{
+		{
+			SourceLabels: model.LabelNames{"__name__"},
+			Regex:        relabel.MustNewRegexp("neuron.*|system_.*|execution_.*"),
+			Action:       relabel.Keep,
+		},
+		{
+			SourceLabels: model.LabelNames{"instance_id"},
+			TargetLabel:  ci.InstanceID,
+			Regex:        relabel.MustNewRegexp("(.*)"),
+			Replacement:  "${1}",
+			Action:       relabel.Replace,
+		},
```

Comment:
Don't need to change if you don't want, but for future reference I saw that you can do this a bit more simply with:
```
{
    Regex: relabel.MustNewRegexp("instance_id"),
    Replacement: ci.InstanceID,
    Action: relabel.LabelMap,
}
```